src/java.desktop/unix/classes/sun/awt/screencast/ScreencastHelper.java

```diff
@@ -0,0 +1,74 @@
+package sun.awt.screencast;
+
+import sun.security.action.GetPropertyAction;
+
+import java.security.AccessController;
+import java.util.prefs.Preferences;
+
+@SuppressWarnings("removal")
+public class ScreencastHelper {
+
+    private static final boolean screencastDebug;
+    private static final boolean isNativeLoaded;
+
+    private static final Preferences TOKEN_STORE =
+            Preferences.userNodeForPackage(ScreencastHelper.class);
+    private static final String TOKEN_KEY = "restoreToken";
+
+    private ScreencastHelper() {}
+
+    static {
+        screencastDebug = Boolean.parseBoolean(
+                AccessController.doPrivileged(
+                        new GetPropertyAction(
+                                "awt.robot.screencastDebug",
+                                "false"
+                        )
+                ));
+
+        isNativeLoaded = loadPipewire(screencastDebug);
+        if (!isNativeLoaded) {
+            System.err.println(
+                    "Could not load native libraries for ScreencastHelper"
+            );
+        }
+    }
+
+    public static boolean isAvailable() {
+        return isNativeLoaded;
+    }
+
+    private static native boolean loadPipewire(boolean screencastDebug);
+
+    public static void resetToken() {
+        TOKEN_STORE.remove(TOKEN_KEY);
+    }
+
+    // called from native
+    private static void storeToken(String token) {
+        TOKEN_STORE.put(TOKEN_KEY, token);
+    }
+
+    private static String getToken() {
+        return TOKEN_STORE.get(TOKEN_KEY, null);
+    }
+
+    private static final int DENIED = -11; //defined in native code
+
+    private static synchronized native int getRGBPixelsImpl(
```

Comment:
This code was written with the understanding that it will not run in parallel, so the `synchronized` is used.

Moreover, we will get a significant benefit from removing `synchronized` only if `getRGBPixelsImpl` is called frequently from different threads. 
If it is called from a single thread(which seems to me to be the main use case), the results of my quick tests with JMH show that the difference between synchronized and not synchronized is within the margin of error.

Removing the `synchronized` keyword would require additional testing, which at this point I want to avoid.
And since this is not a public API we have no problem revisiting it later.

P.S. I removed `synchronized` modifier from `getRGBPixels`

Reply:
> P.S. I removed synchronized modifier from getRGBPixels
My main point was that you don't seem to need to have both, and while I picked one you picked the other ;-) So it's OK.